packages/wallet-adapter-core/src/utils/helpers.ts

```diff
@@ -105,3 +107,48 @@ export const fetchDevnetChainId = async (): Promise<number> => {
   const aptos = new Aptos(); // default to devnet
   return await aptos.getChainId();
 };
+
+/**
+ * A function that will partition the provided wallets into two list — `defaultWallets` and `moreWallets`.
+ * By default, the wallets will be partitioned by whether or not they are installed or loadable.
+ * You can pass your own partition function if you wish to customize this behavior.
+ */
+export function partitionWallets(
+  wallets: ReadonlyArray<AnyAptosWallet>,
+  partitionFunction: (wallet: AnyAptosWallet) => boolean = isInstalledOrLoadable
+) {
+  const defaultWallets: Array<AnyAptosWallet> = [];
+  const moreWallets: Array<AnyAptosWallet> = [];
+
+  for (const wallet of wallets) {
+    if (partitionFunction(wallet)) defaultWallets.push(wallet);
+    else moreWallets.push(wallet);
+  }
+
+  return { defaultWallets, moreWallets };
+}
+
+/** Returns true if the wallet is installed or loadable. */
+export function isInstalledOrLoadable(wallet: AnyAptosWallet) {
+  return (
+    wallet.readyState === WalletReadyState.Installed ||
+    wallet.readyState === WalletReadyState.Loadable
+  );
+}
+
+/**
+ * Returns true if the user is on desktop and the provided wallet requires installation of a browser extension.
+ * This can be used to decide whether to show a "Connect" button or "Install" link in the UI.
+ */
+export function isInstallRequired(wallet: AnyAptosWallet) {
+  const isWalletReady = isInstalledOrLoadable(wallet);
+  const isMobile = !isWalletReady && isRedirectable();
+
+  return !isMobile && !isWalletReady;
+}
+
+/** Truncates the provided wallet address at the middle with an ellipsis. */
+export function truncateAddress(address: string | undefined) {
```

Comment:
lets also update [antd](https://github.com/aptos-labs/aptos-wallet-adapter/blob/main/packages/wallet-adapter-ant-design/src/WalletSelector.tsx#L12) and [mui](https://github.com/aptos-labs/aptos-wallet-adapter/blob/main/packages/wallet-adapter-mui-design/src/WalletButton.tsx#L52C17-L52C32) to import from here

Reply:
I already updated MUI to use it last week. I just pushed a commit to make the same change in Antd. The Antd wallet adapter still needs a lot more refactoring to be at par with the shadcn and MUI wallet selectors, but I'll do that in my next PR.